### PR TITLE
CICD ~ refactor + add PR auto-fixes

### PR DESCRIPTION
**Pull request by rivy**
This is a continuation of #2368.

Style warnings have been converted to errors (with instructions for repair in the error messages; see <https://github.com/rivy-t/rs.coreutils/actions/runs/953446073>). This should push contributors to fix issues earlier and rely less on maintainers to ask for changes.

BusyBox and GNU tests still produce soft warnings when any test fails (with test summaries of failed tests; see <https://github.com/rivy-t/rs.coreutils/actions/runs/953373363> for a fully successful run with only a BusyBox test failure summary).

We should ask for "no-error/no-red PRs" from contributors.

A new 'FixPR' workflow runs on any PR merge to 'master' which auto-fixes 'Cargo.lock' and code formatting, when necessary.

Suggestions for changes?

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
Sounds good! I think it's a good solution to still urge contributors to conform to rustfmt, but to do so automatically and be able to merge it anyway if it's not quite right.

**Comment by sylvestre** [comment]
Excellent
I wonder if it would be possible to show the results in
` GnuTests / Run GNU tests (pull_request) Successful in 74m — Run GNU tests `
like
` codecov/project Successful in 1s — 71.87% (+0.00%) compared to 115eb5e `

File: .github/workflows/CICD.yml [hunk 1]
```diff
@@ -235,8 +247,6 @@ jobs:
           # { os, target, cargo-options, features, use-cross, toolchain }
           - { os: ubuntu-latest  , target: arm-unknown-linux-gnueabihf , features: feat_os_unix_gnueabihf , use-cross: use-cross }
           - { os: ubuntu-latest  , target: aarch64-unknown-linux-gnu   , features: feat_os_unix_gnueabihf , use-cross: use-cross }
-          - { os: ubuntu-latest  , target: x86_64-unknown-linux-gnu    , features: feat_os_unix           , use-cross: use-cross }
```
Comment (jhscheer) [inline .github/workflows/CICD.yml:238]: @rivy I'm missing ubuntu-latest x86_64. Is there a specific reason why you removed it or was it an oversight?

Comment (rivy) [reply .github/workflows/CICD.yml:238]: Looks like a mistake in 298851096ef94. I only meant to remove the single line with the deprecated `ubuntu-16.04` image.

**Pull request merged**